autonomi/src/client/registers.rs

```diff
@@ -147,23 +147,11 @@ impl Client {
                     try_deserialize_record(&record).map_err(|_| RegisterError::Serialization)?;
                 signed_reg
             }
-            // manage forked register case
             Err(NetworkError::GetRecordError(GetRecordError::SplitRecord { result_map })) => {
-                debug!("Forked register detected for {address:?} merging forks");
-                let mut registers: Vec<SignedRegister> = vec![];
-                for (_, (record, _)) in result_map {
-                    registers.push(
-                        try_deserialize_record(&record)
-                            .map_err(|_| RegisterError::Serialization)?,
-                    );
-                }
-                let register = registers.iter().fold(registers[0].clone(), |mut acc, x| {
-                    if let Err(e) = acc.merge(x) {
-                        warn!("Ignoring forked register as we failed to merge conflicting registers at {}: {e}", x.address());
-                    }
-                    acc
-                });
-                register
+                error!("Got split record error for register at address: {address}. This should've been handled at the network layer");
+                Err(RegisterError::Network(NetworkError::GetRecordError(
+                    GetRecordError::SplitRecord { result_map },
```

Comment:
Register is always `merged` if split detected.
And, due to the fact that register-record holds list of ops, it might be possible the get_record will report with split.

I don't think here a merge can be skipped , neither shall the error to be raised up ?

also, as in this PR, the `merge` is moved into `network` level, this means on client level here, we shall never receive an error of split ?
In that case, we shall raise/log something more serious ?

Reply:
Yeah, we'd not get this error on the client side. I have just noted this down for future reference.